packages/driver/src/cy/commands/actions/attachFile.ts

```diff
@@ -6,23 +6,111 @@ import $errUtils from '../../../cypress/error_utils'
 import $actionability from '../../actionability'
 import { addEventCoords, dispatch } from './trigger'
 
-const createFileList = (files: Cypress.FileReferenceObject[]): FileList => {
+/* dropzone.js relies on an experimental, nonstandard API, webkitGetAsEntry().
+ * https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry
+ * The behavior here varies between browsers, and though unpleasant, we have to attempt
+ * to replicate it as far as possible.
+ *
+ * In Firefox, attempting to set webkitGetAsEntry() fails - but this is fine, because
+ * Firefox also returns a useful value from item.webkitGetAsEntry().
+ *
+ * In webkit browsers however, item.webkitGetAsEntry() returns null, because our File()
+ * instances don't actually correspond with a file on disk. But the property is writeable,
+ * so we're able to override it with our own implementation that returns a proper object.
+ */
+const tryMockWebkit = (item) => {
+  try {
+    item.webkitGetAsEntry = () => {
+      return {
+        isFile: true,
+        file: (callback) => callback(item.getAsFile()),
+      }
+    }
+  } catch (e) {
+    // We're in Firefox, this is fine.
```

Comment:
Can we validate the browser is actually firefox before swallowing the error? Cypress.config('browsers') will return the current browser.

Reply:
Discussed in more detail below, but I don't think checking the browser here adds value.